third_party/coro/coro.c

```diff
@@ -103,13 +105,29 @@ coro_init (void)
 
   coro_transfer (new_coro, create_coro);
 
-#if __GCC_HAVE_DWARF2_CFI_ASM && __amd64
-  asm (".cfi_undefined rip");
-#endif
-
-  func ((void *)arg);
-
-  /* the new coro returned. bad. just abort() for now */
+  __asm__(
+	  /*
+	   * Call-chain ends here: we need to invalidate this frame's return
+	   * address to make unwinding stop here.
+	   * Some nuances on x86_64, see:
+	   * https://github.com/libunwind/libunwind/blob/ec171c9ba7ea3abb2a1383cee2988a7abd483a1f/src/x86_64/Gstep.c#L91-L92
```

Comment:
Please don't leave links to other comments in the comment, just add a comment you want. Also, it would be nice to place every comment near the corresponding line of assembly.

Reply:
I  don't agree with you on this point: this comment contains references to the library unwinding is based on, so that one can understand what assumptions we rely on.